docs/workflow/building/libraries/cross-building.md

```diff
@@ -24,108 +23,102 @@ and for arm64 you need:
-=======================================
+Note that by default ILLinker trimming is enabled and libraries built above for arm can't be used on other arches. To build architecture-independent managed part of libraries for arm using subset:
 
-It is possible to build corefx binaries (native and managed) for the Linux ARM Emulator (latest version provided here: [#5394](https://github.com/dotnet/runtime/issues/5394)).
-The `scripts/arm32_ci_script.sh` script does this.
+    $ ./build.sh --arch arm --librariesConfiguration Release --subset libs.sfx /p:ILLinkTrimAssembly=false
 
-The following instructions assume that:
-* You have set up the extracted emulator at `/opt/linux-arm-emulator` (such that `/opt/linux-arm-emulator/platform/rootfs-t30.ext4` exists)
-* The mount path for the emulator rootfs is `/opt/linux-arm-emulator-root` (change this path if you have a working directory at this path).
+Build artifacts can be found in `artifacts/bin/microsoft.netcore.app.runtime.<TargetOS>-<BuildArch>/<BuildType>/runtimes/<TargetOS>-<BuildArch>/lib/net7.0/`. For more details on the build configurations see [project-guidelines](/docs/coding-guidelines/project-guidelines.md).
 
-All the following instructions are for the Release mode. Change the commands and files accordingly for the Debug mode.
+Both native and managed parts can be built at the same time with:
 
-To just build the native and managed corefx binaries for the Linux ARM Emulator, run the following command:
-```
-prajwal@ubuntu ~/corefx $ ./scripts/arm32_ci_script.sh \
-    --emulatorPath=/opt/linux-arm-emulator \
-    --mountPath=/opt/linux-arm-emulator-root \
-    --buildConfig=Release
-```
+    $ ROOTFS_DIR=`pwd`/.tools/rootfs/arm ./build.sh --cross --arch arm --librariesConfiguration Release --subset libs.native+libs.sfx
 
-The Linux ARM Emulator is based on the soft floating point and thus the native binaries are generated for the armel architecture. The corefx binaries generated by the above command can be found at `~/corefx/artifacts/bin/Linux.armel.Release`, `~/corefx/artifacts/bin/Linux.AnyCPU.Release`, `~/corefx/artifacts/bin/Unix.AnyCPU.Release`, and `~/corefx/artifacts/bin/AnyOS.AnyCPU.Release`.
+Build libraries for a new architecture
+===================================
 
+When building for a new architecture you will need to build the native pieces separately from the managed pieces in order to correctly boot strap the native runtime.
 
-Build corefx for a new architecture
-===================================
+Native part build for target architecture:
 
-When building for a new architecture you will need to build the native pieces separate from the managed pieces in order to correctly boot strap the native runtime. Instead of calling build.sh directly you should instead split the calls like such:
+    $ ROOTFS_DIR=`pwd`/.tools/rootfs/<BuildArch> ./src/native/libs/build-native.sh -release -<BuildArch> -cross -outconfig net7.0-Linux-Release-<BuildArch>
```

Comment:
`outconfig` is unnecessary.

Reply:
For arm build without `outconfig` default dir is `Linux-Release-arm`, while build using `build.sh libs.native` creates `net7.0-Linux-Release-arm` dir. Also, managed build fails after native build without correct `outconfig` with:
```
/home/runtime/.dotnet/sdk/7.0.100-rc.1.22431.12/Sdks/NuGet.Build.Tasks.Pack/build/NuGet.Build.Tasks.Pack.targets(221,5): error NU5026: The file '/home/runtime/artifacts/bin/native/net7.0-Linux-Release-arm/libSystem.IO.Ports.Native.so.dbg' to be packed was not found on disk. [/home/runtime/src/libraries/System.IO.Ports/pkg/runtime.linux-arm.runtime.native.System.IO.Ports.proj]
  System.Runtime.Serialization.Schema -> /home/runtime/artifacts/bin/System.Runtime.Serialization.Schema/Release/net7.0/System.Runtime.Serialization.Schema.dll
```

Reply:
Usage of `eng/native/libs/build-runtime.sh` directly is necessary when there is no msbuild, which is not the case when we are cross compiling from linux host. The recommended way is `ROOTFS_DIR=/path ./build.sh libs.native -cross # with -arch and/or -os args`.

Invoking `eng/native/libs/build-runtime.sh`  (without -outconfig) followed by `build.sh libs.managed -cross` fails, but if you are going to use `build.sh libs.managed`, why not use `build.sh libs.native` in the first place? 🙂

Note that there is no CI pipeline testing build-native directly.

Nonetheless, since outconfig without TFM is guaranteed to fail, I think we should stop accepting this argument:

```diff
+ __CurrentFramework=net7.0
...
-        outconfig|-outconfig)
-            __outConfig="$2"
-            __ShiftArgs=1
-            ;;
...
- __outConfig="${__outConfig:-"$__TargetOS-$__TargetArch-$__BuildType"}"
+ __outConfig="$__CurrentFramework-$__TargetOS-$__TargetArch-$__BuildType"
```
with every new version (once a year), we update tfm in various places including `eng/common/tools.sh`, so updating it won't be needed.

Reply:
Sure, if msbuild is available it's easier to just build both subsets at the same time, I have removed `build-native.sh` usage